core/src/main/scala/kafka/log/Log.scala

```diff
@@ -75,7 +75,7 @@ case class LogAppendInfo(var firstOffset: Long,
  *
  */
 @threadsafe
-class Log(val dir: File,
+class Log(@volatile var dir: File,
```

Comment:
I identified one harmless case of concurrent read access for logging. I eliminated the concurrent access by updating the log message in way that does not access the dirName.